tests/components/homematicip_cloud/test_hap.py

```diff
@@ -140,6 +145,10 @@ async def test_hap_create(
             assert await hap.async_setup()
 
 
+@patch(
+    "homeassistant.components.homematicip_cloud.hap.ConnectionContextBuilder.build_context_async",
+    return_value=ConnectionContext(),
+)
 async def test_hap_create_exception(
     hass: HomeAssistant, hmip_config_entry: MockConfigEntry, mock_connection_init
 ) -> None:
```

Comment:
The patch decorator can be quite tricky as it automatically binds one argument. As no additional ones had been added, it replaced one fixture which causes a lot of RuntimeWarnings and possibly wrong test execution.

I'd recommend to stick with the patch contextmanager as that's a bit more obvious. For a fix, see #142961